cmd/agentbasedinstaller/register.go

```diff
@@ -103,6 +104,7 @@ func RegisterCluster(ctx context.Context, log *log.Logger, bmInventory *client.A
 
 	annotations := aci.GetAnnotations()
 	if installConfigOverrides, ok := annotations[controllers.InstallConfigOverrides]; ok {
+		var reJsonField = regexp.MustCompile(`(?i)"([^"]*(password)[^"]*)":\s*"[^\,]*"`)
```

Comment:
I think the part that matches the password itself, `[^\,]*` might instead need to be something like `(\\{2}|\\"|[^"])*`

Alternatively, it might be easier to parse the JSON and then redact the fields, then reserialise, than to use a regex.

Reply:
Changed regex.

Reply:
I assume you tested this, because I definitely did not :slightly_smiling_face: